docs/zq/functions/flatten.md

```diff
@@ -0,0 +1,25 @@
+### Function
+
+&emsp; **flatten** &mdash; transform a record into a flattened map
+
+### Synopsis
+
+```
+flatten(val: record) -> |{[string]:<any>}|
+```
+### Description
+The _flatten_ function returns a map of the flattened key/values of the record
```

Comment:
```suggestion
The _flatten_ function returns a map from flattened key to value for the record
```

Reply:
In reworking this for clarification, I just realized flatten's use of a map doesn't provide a way to preserve field order of a record.  I think we should we should use `[{key:[string],value:any}]` instead here.  For now, I updated the docs and created issue #3615.